### ISSUE TITLE
`Unexpected token` error caused by dynamically inserted `import` statement

### ISSUE THREAD

**Issue by bertdeblock** [embroider-build/embroider#1304]
When updating all Embroider packages to v2, I saw the following error in CI:

```
ERROR in ./tests/integration/components/dropdown/actions/link-test.js 676:13
Module parse failed: Unexpected token (676:13)
File was processed with these loaders:
 * ../../../builds/project/manuport/manuport-customer-portal-frontend/node_modules/thread-loader/dist/cjs.js
 * ../../../builds/project/manuport/manuport-customer-portal-frontend/node_modules/@embroider/babel-loader-8/index.js
You may need an additional loader to handle the result of these loaders.
| cov_w9lbfd2jg();
| import { createTemplateFactory } from "../../../../../../externals/@ember/template-factory";
> import link, link0 from "../../../../../helpers/link.js";
| import DropdownActionsLink from "../../../../../components/dropdown/actions/link/index.js";
| import { module, test } from 'qunit';
```

This error happened for an integration test that uses the `{{link}}` helper coming from [`ember-link`](https://github.com/buschtoens/ember-link):

```js
import { module, test } from 'qunit';
import { setupRenderingTest } from 'cuspor/tests/helpers';
import { click, render } from '@ember/test-helpers';
import { hbs } from 'ember-cli-htmlbars';
import { setupLink, linkFor } from 'ember-link/test-support';

module('Integration | Component | dropdown/actions/link', function (hooks) {
  setupRenderingTest(hooks);
  setupLink(hooks);

  test('it calls the Link API', async function (assert) {
    const link = linkFor('some.route'); // NOTE: Local variable named `link`.
    link.onTransitionTo = assert.step('link clicked');

    await render(hbs`
      <Dropdown::Actions::Link @link={{link "some.route"}}>
        Link
      </Dropdown::Actions::Link>
    `);

    await click('[data-test-dropdown-actions-link]');

    assert.verifySteps(['link clicked']);
  });
});
```

The test also had a local variable declared named `link`.
If I renamed the local variable to something else, the error disappeared.

Looking at the generated `import` statement:

```js
import link, link0 from "../../../../../helpers/link.js";
```

It seems Embroider noticed the local variable (hence the `link0`?), but I guess `link,` was supposed to be removed?

**Comment by ef4** [comment]
I think this might be fixed by #1308 although I was not able to make your example fail in exactly the same way.

I think this could have been caused by the combination of:
 - our transform adds lexically scoped arguments to templates
 - our transform didn't correctly respect *existing* lexically scoped arguments to templates
 - depending on ones full set of of babel plugins, our transform might end up reprocessing some of its own output, which would mean the lexically scoped arguments we added are now "existing" ones that we didn't respect correctly.

**Comment by bertdeblock** [comment]
Going to close for now, haven't run into any similar issues.
